### ISSUE TITLE
I can download CSV files

### ISSUE THREAD

**Issue by pmackay** [folklabs/connecting-communities#4]
**story**
As a council officer, I can access the system to download a CSV file containing contact details and email addresses so that I can send an email from Outlook to all the groups in my search result.

**Comment by pmackay** [comment]
@GilesGibson @GreenCChampion I'd like to unpick this a bit more. If downloading CSVs is just to be able to send via email software, would something different bring greater benefits? What if there was a system to manage contacts, groups and email message sending?

**Comment by djwesto** [comment]
I was really excited when @pmackay raised this over the weekend – @GreenCChampion are you aware of any existing CRM (=Customer Relationship Management) system within Lambeth to manage community groups? I assume there isn't one but this could be an opportunity to put something in place that council officers could use to each manage their own contacts.

**Comment by GilesGibson** [comment]
I would avoid trying to build a CRM system. That is not the target market and there are plenty of those already out there.

At a later stage then maybe looking at an API to transfer contacts over may be worth it.

As soon as you go down the road of a CRM system then it becomes an internal system for officers and pivots drastically from our aims.

**Comment by pmackay** [comment]
@GilesGibson but this need is about "downloading CSV files". As I user I dont want to do that, surely? Its a patch for a bigger goal..

**Comment by GilesGibson** [comment]
yes, downloading a filtered/sorted list is needed for phase 1. It is the additional information about history of communications that is more CRM.

**Comment by pmackay** [comment]
I'm saying its not a great user experienece anyway to have to download a CSV file...

**Comment by GilesGibson** [comment]
True, but we have limited time in the first phase.

**Comment by pmackay** [comment]
What would be the next phase, in an ideal world?

**Comment by GilesGibson** [comment]
expand the source sites (Project Dirt, Lambeth Young Coop etc). Improve ability to edit on the site. Improve ability to add datasets.Widget to embed in other web sites. Ability for community group sites to search a wider area.

Loads more as discussed.

**Comment by pmackay** [comment]
So its up to council people to merge their contacts lists once they've downloaded a CSV?

**Comment by djwesto** [comment]
For me, one advantage that a CRM system might offer would be greater level of detailed permissions access for council officers – at present it's not clear to me how we'd meet @GreenCChampion's requirement that each officer can edit "their own contacts" – a CRM might bring in tools for managing contacts owned by multiple people, if some contacts belong to more than one officer at a time.

**Comment by GilesGibson** [comment]
I suppose I saw it that for the first phase, we would get the contact lists, clean them up manually (if necessary) and then upload them to Ohana. The officers could then access Ohana, do a search and filter and then extract a new csv list that they could add to their Outlook. 

In later versions then some sort of "ownership" of contacts may be necessary - an audit trail of what was done by whom and when.

**Comment by GilesGibson** [comment]
I just think that at this stage, and many of the next ones too, we don't want to be creating yet another CRm system for local gov. Every local authority has probably started to use one system or another. To try and get them to change or run two side by side is not great. Giving them the list of contacts that they may want to import (or via an API) into whatever system they run is for now a workaround.

**Comment by GreenCChampion** [comment]
I think that for the time being Ohana or whatever it is has to be an amalgamation of what other people are doing/using - rather than a thing that needs servicing in its own right. We can advise people how to collect/store data so that it can be uploaded and shared. This will be particularly important for people just starting to do asset maps - eg in Health

**Comment by pmackay** [comment]
How important is it that CSV download is restricted to authorized people?

**Comment by GreenCChampion** [comment]
I hate the idea that officers will be downloading CSV files and sending out mass emails so I would say it is very important. this is required for the GIFTS replacement element of the work but for the most part we want people to use networks. They will get  a much better response on an issue if they ask network owners to speak to their networks rather than if they email direct. So I would say it is very important.

**Comment by GilesGibson** [comment]
At present, officers can access GIFTS and spam everyone, no gatekeeping involved really. As a recipient of this SPAM I understand the reasons (naff systems at the back end) and delete accordingly.

If we can track ownership of content then this will help. Not sure however how we share the ability to send info out. In theory it should be on an opt-in level.For those groups that are on sites that are linked with Connecting Communities then they can easily control what they get, for those aggregated from elsewhere (GIFTS, officer lists) then some sort of gatekeeping/opt in/out system would be prudent.

**Comment by pmackay** [comment]
We should just focus on the GIFTS data to start with, which is a list of
organisation contact details (not individuals). So is it ok if that can be
downloaded via CSV by anyone?

**Comment by GilesGibson** [comment]
Be nice to think about how we could regulate officer spamming.

**Comment by pmackay** [comment]
Am just aware that authenticated/restricted access to content will be
tricky with the Ohana front end. Its not supported at all currently.

**Comment by GilesGibson** [comment]
@GreenCChampion how would you want to moderate the Greenchampions list, if that was uploaded?

**Comment by GreenCChampion** [comment]
I think I would want people to ask me to send out info on their behalf, however, if some on my list were orgs. rather than individuals, perhaps I would be happy to share. Also then I would know about other orgs that I should be in touch with. 
I am thinking of another list i hold - which is food banks. Their email addresses are published on their websites so I don't own them - anyone can contact them - but they are just more likely to respond if I contact them because they know me (they often complain they got approached by lots of people from the council and so i try to coordinate contact from the council).

**Comment by pmackay** [comment]
Just to be clear, I am assuming in this initial phase we're talking about organisation emails, not individuals. I appreciate in some cases some grassroots orgs will just have people's personal emails for contacts. 

The Ohana model is that the information is important for any citizen and open to all, via the API and web UI. So I think this whole issue of protected email lists and so on needs more discussion to understand how to do it properly in the context of LCC.

**Comment by djwesto** [comment]
Can I clarify where we've reached on downloading CSVs? We're describing a lot of functions that sound very much like an advanced CRM:
- More than one Council Officer can log into the system
- Each Council Officer is allowed to each have their own personal login
- Each Officer can upload their own set of records which they add to the system as some sort of file upload or by manually keying them into the system
- Each Officer can access a control which allows them to search for records in the system
- Each Officer can change their view of all the contact records available to them by setting "filters" on the display by selecting various category terms or other aspects of the records
- Once the view has been filtered to their liking, each Officer can then choose to download all the filtered records as a CSV file

If this is all correct, I have some questions:
- Can I, as Council Officer Chris **only** filter and search _my own_ contact records or everybody's?
- If I perform a search and the search returns a bunch of other Officers' contact records, can I then download a CSV containing their records or _only_ mine?

All these questions and more need answers and we're beginning to suspect that the solution to these requirements is some sort of CRM system.

If, as @GilesGibson suggested already, we are _not_ to build a CRM system as part of this project (we couldn't do so for this project budget), then we need to know what Lambeth already uses as a CRM system internally?

@GreenCChampion is there anyone we could possibly ask about this?

**Comment by GreenCChampion** [comment]
We don't have a CRM system...

**Comment by djwesto** [comment]
Literally, nobody in the whole of Lambeth Council is running any sort of CRM system of any kind? Not in the Comms team or in any of the technical teams which require large-scale contact management?

**Comment by GreenCChampion** [comment]
Planning does... 
Events does..
but they are all separate

**Comment by djwesto** [comment]
How many of the departments which will be using the system we're proposing to build (if this prototype is approved) are already using a CRM?

In case it seems a little off-track, the main reason I'm asking all these questions is because we can't build a complete CRM into Ohana and we couldn't build one from scratch. But, if the Officers who'd be using the system already have a CRM, we might be able to connect the data in our system into their CRM so they could then perform the complex manipulations there instead.

**Comment by GreenCChampion** [comment]
Can we leave this for now? I am worried we are focusing too much on council officers and we should be building something that is as useful to the community..
